backend/src/main/java/ai/verta/modeldb/project/FutureProjectDAO.java

```diff
@@ -863,51 +863,36 @@ public InternalFuture<List<GetResourcesResponseItem>> deleteProjects(List<String
-            .setService(ServiceEnum.Service.MODELDB_SERVICE)
-            .setModeldbServiceAction(modelDBServiceActions)
-            .build();
-    GetSelfAllowedResources getAllowedResourcesRequest =
-        GetSelfAllowedResources.newBuilder()
-            .addActions(action)
-            .setResourceType(
-                ResourceType.newBuilder()
-                    .setModeldbServiceResourceType(modelDBServiceResourceTypes))
-            .setService(ServiceEnum.Service.MODELDB_SERVICE)
-            .build();
-    return FutureGrpc.ClientRequest(
-            uac.getAuthzService().getSelfAllowedResources(getAllowedResourcesRequest), executor)
+  private InternalFuture<Collection<String>> getSelfAllowedResources(
+      ModelDBServiceActions modelDBServiceActions, List<String> requestedResourcesIds) {
+    return uacApisUtil
+        .getAllowedEntitiesByResourceType(
+            modelDBServiceActions, ModelDBResourceEnum.ModelDBServiceResourceTypes.PROJECT)
         .thenApply(
             getAllowedResourcesResponse -> {
               LOGGER.trace(CommonMessages.ROLE_SERVICE_RES_RECEIVED_MSG);
               LOGGER.trace(
                   CommonMessages.ROLE_SERVICE_RES_RECEIVED_TRACE_MSG, getAllowedResourcesResponse);
-
-              var resourcesIds = new ArrayList<String>();
-              if (!getAllowedResourcesResponse.getResourcesList().isEmpty()) {
-                for (Resources resources : getAllowedResourcesResponse.getResourcesList()) {
-                  resourcesIds.addAll(resources.getResourceIdsList());
-                }
-                // Validate if current user has access to the entity or not
-                // resourcesIds.retainAll(requestedResourcesIds);
-                if (requestedResourcesIds.isEmpty()) {
-                  return resourcesIds;
-                } else if (resourcesIds.containsAll(requestedResourcesIds)) {
-                  return requestedResourcesIds;
-                } else {
-                  for (var requestedId : requestedResourcesIds) {
-                    if (!resourcesIds.contains(requestedId)) {
-                      resourcesIds.remove(requestedId);
-                    }
-                  }
+              boolean allowedAllResources =
+                  RoleServiceUtils.checkAllResourceAllowed(getAllowedResourcesResponse);
+              if (allowedAllResources) {
+                return new ArrayList<>(requestedResourcesIds);
+              }
+              Set<String> allowedProjectIds =
+                  RoleServiceUtils.getResourceIds(getAllowedResourcesResponse);
+              // Validate if current user has access to the entity or not
+              // resourcesIds.retainAll(requestedResourcesIds);
+              if (requestedResourcesIds.isEmpty()) {
+                return allowedProjectIds;
+              } else if (allowedProjectIds.containsAll(requestedResourcesIds)) {
+                return requestedResourcesIds;
+              }
```

Comment:
one more redundant else here. ;)

Reply:
I don't understand how it is reduntant. It has `if` in it

Reply:
line `886` returns, so if the return happens, the `else` will always execute. Hence, you don't need the `else`...just the next `if`.

Reply:
removed